compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -293,7 +294,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 self.check_lang_item_path(lang_item, expr, hir_id)
             }
             ExprKind::Path(ref qpath) => self.check_expr_path(qpath, expr, &[]),
-            ExprKind::InlineAsm(asm) => self.check_expr_asm(asm),
+            ExprKind::InlineAsm(asm) => {
+                self.deferred_asm_checks.borrow_mut().push((asm, expr.hir_id));
```

Comment:
Why defer and not check it directly?

Reply:
Because it may contain unresolved inference variables, especially in the output. We need to wait for fallback to resolve all inference variables in the input, too